app/src/org/commcare/dalvik/activities/CallOutActivity.java

```diff
@@ -107,16 +110,25 @@ public void onCancel(DialogInterface dialog) {
     }
     
     private void dispatchAction(String action) {
+        // using createChooser to handle any errors gracefully
         if(Intent.ACTION_CALL.equals(action) ) {
             tManager.listen(listener, PhoneStateListener.LISTEN_CALL_STATE);
             
             Intent call = new Intent(Intent.ACTION_CALL);
             call.setData(Uri.parse("tel:" + number));
-            startActivity(call);
+            if(call.resolveActivity(getPackageManager()) != null){
+                startActivity(call);
+            } else {
+                Toast.makeText(getApplicationContext(), Localization.get("callout.failure.dialer"), Toast.LENGTH_SHORT).show();

```

Comment:
In general when inside of an Activity, you should use the current activity as the Context argument, so any details about it can be properly passed along to the next layer.